test/units/TEST-88-UPGRADE.sh

```diff
@@ -52,12 +52,31 @@ check_sd() {
         fail=1
     fi
 
+    systemctl status upgrade_timer_test.{service,timer}
+    timer1_new=$(systemctl show -P TimersCalendar upgrade_timer_test.timer)
+    timer2_new=$(systemctl show -P NextElapseUSecRealtime upgrade_timer_test.timer)
+
+    if [[ $timer1 != "$timer1_new" ]]; then
+        echo "Timer changed unexpectedly: '$timer1' != '$timer1_new'"
+        fail=1
+    fi
+    if [[ $timer2 != "$timer2_new" ]]; then
+        echo "Timer changed unexpectedly: '$timer2' != '$timer2_new'"
+        fail=1
+    fi
+
     [[ $fail -eq 0 ]]
 }
 
 # Copy the unit in /run so systemd finds it after the downgrade
 cp /usr/lib/systemd/tests/testdata/units/TEST-88-UPGRADE.service /run/systemd/system
 
+now=$(date +%s)
+after_2h=$((now + 3600 * 2))
+systemd-run --on-calendar=@$after_2h -u upgrade_timer_test date
```

Comment:
I do not know why it is not warned by the liner, but I think it is better to quote the string with variable:
```sh
--on-calendar="@$after_2h"
```